plutus-ledger-api/src/v1/interval.rs

```diff
@@ -82,6 +103,122 @@ where
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum TryFromPlutusIntervalError {
+    #[error("Interval is invalid.")]
+    InvalidInterval,
+    #[error("Interval with open bound could not be converted.")]
+    UnexpectedOpenBound,
+}
+
+impl<T> TryFrom<PlutusInterval<T>> for Interval<T>
+where
+    T: FeatureTraits,
+{
+    type Error = TryFromPlutusIntervalError;
+
+    fn try_from(interval: PlutusInterval<T>) -> Result<Self, Self::Error> {
+        Ok(match interval {
+            PlutusInterval {
+                from:
+                    LowerBound {
+                        bound: Extended::Finite(start),
+                        closed: lc,
+                    },
+                to:
+                    UpperBound {
+                        bound: Extended::Finite(end),
+                        closed: uc,
+                    },
+            } => {
```

Comment:
Should we ensure that `start < end` always hold true?

Reply:
I added this check